src/main/java/com/google/api/codegen/ruby/RubyDiscoveryProvider.java

```diff
@@ -0,0 +1,65 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.ruby;
+
+import com.google.api.Service;
+import com.google.common.collect.Multimap;
+import com.google.protobuf.Api;
+import com.google.protobuf.Method;
+
+import com.google.api.codegen.ApiaryConfig;
+import com.google.api.codegen.DiscoveryProvider;
+import com.google.api.codegen.GeneratedResult;
+import com.google.api.codegen.SnippetDescriptor;
+
+import java.io.IOException;
+
+/**
+ * A DiscoveryProvider for generating Ruby fragments.
+ */
+public class RubyDiscoveryProvider implements DiscoveryProvider {
+  private final RubyDiscoveryContext context;
+  private final RubyProvider provider;
+
+  public RubyDiscoveryProvider(Service service, ApiaryConfig apiaryConfig) {
+    this.context = new RubyDiscoveryContext(service, apiaryConfig);
+    this.provider = new RubyProvider();
+  }
+
+  @Override
+  public GeneratedResult generateFragments(Method method, SnippetDescriptor snippetDescriptor) {
+    return provider.generate(method, snippetDescriptor, context);
+  }
+
+  @Override
+  public void output(String outputPath, Multimap<Method, GeneratedResult> methods)
+      throws IOException {
+    Api api = getService().getApis(0);
```

Comment:
Is this the code the same in all languages? In that case we should have this method defined in a common ancestor. If that's complicated because we need to define such an ancestor, we can do that in a follow-up change.